wave_lang/kernel/lang/wave_types.py

```diff
@@ -138,6 +138,9 @@ def __class_getitem__(
             raise TypeError(f"Expected at least 2 arguments, got: {shape_and_dtype}")
 
         shape = shape_and_dtype[:-1]
+        # allow shape to be provided as a tuple instead of as individual elements, to work around lack of unpacking in subscripts for Python 3.10
+        if len(shape) == 1 and isinstance(shape[0], tuple):
+            shape = shape[0]
```

Comment:
Is there a LIT test that show case this use case?

Reply:
This is used in the new `infer_broadcast_shape`.  IE the type is `Register[broadcast_shape, dtype]`, because old versions of python still don't support `Register[*broadcast_shape, dtype]`.